src/Domain/Item/ItemService.ts

```diff
@@ -19,14 +21,23 @@ import { SaveItemsResult } from './SaveItemsResult'
 export class ItemService implements ItemServiceInterface {
   private readonly DEFAULT_ITEMS_LIMIT = 100000
   private readonly SYNC_TOKEN_VERSION = 2
-  private readonly MIN_CONFLICT_INTERVAL_MICROSECONDS = Time.MicrosecondsInAMillisecond
 
   constructor (
     @inject(TYPES.ItemRepository) private itemRepository: ItemRepositoryInterface,
     @inject(TYPES.Timer) private timer: TimerInterface
   ) {
   }
 
+  async computeIntegrityHash(userUuid: string): Promise<string> {
+    const timestampsInMicroseconds = await this.itemRepository.findDatesForComputingIntegrityHash(userUuid)
+
+    const timestampsInMilliseconds = timestampsInMicroseconds.map(timestampsInMicrosecond => Math.floor(timestampsInMicrosecond / Time.MicrosecondsInAMillisecond))
```

Comment:
`timestampsInMicroseconds.map(timestampsInMicrosecond ` should be `timestampsInMicroseconds.map(timestampInMicroseconds`

So singularize mapped variable name to avoid confusion.